solidity/contracts/Oracle.sol

```diff
@@ -139,14 +156,23 @@ contract Oracle is ChainlinkRequestInterface, OracleInterface, Ownable {
     return withdrawableTokens.sub(ONE_FOR_CONSISTENT_GAS_COST);
   }
 
-  function cancel(bytes32 _requestId)
-    external
-  {
-    require(msg.sender == callbacks[_requestId].addr, "Must be called from requester");
-    require(callbacks[_requestId].cancelExpiration <= now, "Request is not expired");
-    Callback memory cb = callbacks[_requestId];
-    require(LINK.transfer(cb.addr, cb.amount), "Unable to transfer");
-    delete callbacks[_requestId];
+  function cancel(
+    bytes32 _requestId,
+    uint256 _payment,
+    bytes4 _callbackFunc,
+    uint256 _expiration
+  ) external {
+    bytes32 paramsHash = keccak256(
+      abi.encodePacked(
+        _payment,
+        msg.sender,
+        _callbackFunc,
+        _expiration)
+    );
+    require(paramsHash == commitments[_requestId], "Params do not match request ID");
+    require(_expiration <= now, "Request is not expired");
+    require(LINK.transfer(msg.sender, _payment), "Unable to transfer");
+    delete commitments[_requestId];
```

Comment:
I realize we had the same order with the previous cancel implementation, but I notice now that we should delete the commitment before transferring LINK to fit with the checks effects interactions pattern.